[all] Use crypto random interface to generate UUID

mlohvynenko opened the pull request.

mlohvynenko committed:
gitignore: ignore cmake user presets

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

MykolaSuperman commented on src/iam/permhandler/permhandler.cpp:
```diff
@@ -108,16 +121,22 @@ InstancePermissions* PermHandler::FindByInstanceIdent(const InstanceIdent& insta
 
-    } while (FindBySecret(newSecret) != mInstancesPerms.end());
+    result = uuid::UUIDToString(uuid);
 
-    return newSecret;
+    return result;
```
maybe just 
```
return uuid::UUIDToString(uuid);
```

mlohvynenko replied:
uuid static string has less buffer size then secret, so compilation fails with the proposed approach
but works fine with an explicit constructor call

`return StaticString<cSecretLen>(uuid::UUIDToString(uuid));`

al1img commented on include/aos/common/tools/uuid.hpp:
```diff
@@ -11,6 +11,10 @@
 #include "aos/common/tools/string.hpp"
 
+namespace aos::crypto {
+class RandomItf;
```
Why not just include "crypto.hpp"?

mykola-kobets-epam replied:
probably better create a dedicated hpp for RandomItf

al1img commented on include/aos/common/tools/uuid.hpp:
```diff
@@ -31,9 +35,9 @@ using UUID = StaticArray<uint8_t, cUUIDSize>;
+ * @return RetWithError<UUID>.
  */
-UUID CreateUUID();
+RetWithError<UUID> CreateUUID(class crypto::RandomItf& randomProvider);
```
We have `CreateUUIDv5` crypto provider method. 

I suggest moving this method there as well. In any case user CreateUUID requires additional interface (randomitf or cryptoprovideritf). Each crypto provider has its own mechanism to generate UUID. Let's use this approach.

mlohvynenko replied:
Make sense, but enhancing this interface doesn't seem right to me, at least from ProviderItf doc.
Probably adding UUIDProviderItf and implementing it by OpenSSLCryptoProvider/MbedTLSCryptoProvider classes 
is a better approach

```
/**
 * Provides interface to manage certificate requests.
 */
class ProviderItf {
public:
```

mlohvynenko committed:
common: crypto: add crypto provider interface

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
all: create UUIDs via crypto provider interface

This patch removes CreateUUID free function and
replaces it with calls to the crypto provider interface.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img reviewed (approved):
`Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>`
